Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs

```diff
@@ -70,10 +67,11 @@ public class CSharpKernel :
 
         public CSharpKernel() : base(DefaultKernelName)
         {
-            _dependencies = new Lazy<DependencyProvider>(GetDependencyProvider);
-
+            _packageRestoreContext = new Lazy<PackageRestoreContext>(() => new PackageRestoreContext());
             RegisterForDisposal(() =>
             {
+                _packageRestoreContext?.Value?.Dispose();
```

Comment:
We should check if the value was created because this will initialize the `Lazy` even if it had not been initialized, and then dispose it.

Reply:
The cleanest way to do this is actually to register it for disposal inside the `Lazy`'s initialization lambda.